cmd/kubeadm/app/phases/certs/certlist.go

```diff
@@ -397,3 +416,58 @@ func setCommonNameToNodeName() configMutatorsFunc {
 		return nil
 	}
 }
+
+// LeafCertificates returns non-CA certificates from the supplied Certificates.
+func LeafCertificates(c Certificates) (Certificates, error) {
+	certTree, err := c.AsMap().CertTree()
+	if err != nil {
+		return nil, err
+	}
+
+	var out Certificates
+	for _, leafCertificates := range certTree {
+		out = append(out, leafCertificates...)
+	}
+	return out, nil
+}
```

Comment:
I moved this to a separate function where it can be unit tested separately from the complex and slow key and CSR generation.